MovieApp/DataLayer/Clients/MovieClient/MovieClient.swift

```diff
@@ -0,0 +1,26 @@
+import Foundation
+
+class MovieClient: MovieClientProtocol {
+
+    private let baseApiClient: BaseApiClient
+
+    init(baseApiClient: BaseApiClient) {
+        self.baseApiClient = baseApiClient
+    }
+
+    func fetchPopularMovies(completionHandler: @escaping (Result<PopularMoviesModel, RequestError>) -> Void) {
+        let queryParameters = [
+            "language": "en-US",
+            "page": "1",
+            "api_key": baseApiClient.apiKey
```

Comment:
Clients should not be aware of the API key. It should be added in BaseApiClient.